generated/xamarin-flexible-sync/realm-todo-dotnet/TaskPage.xaml.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Linq;
+using RealmTemplateApp.Models;
+using Realms;
+using Xamarin.Forms;
+using System.ComponentModel;
+using Realms.Sync;
+using System.Collections.Generic;
+using Realms.Sync.Exceptions;
+
+namespace RealmTemplateApp
+{
+    public partial class TaskPage : ContentPage
+    {
+        private Realm taskRealm;
+        private User user;
+        private IEnumerable<Task> _tasks;
+
+        public TaskPage()
+        {
+            InitializeComponent();
+            user = App.RealmApp.CurrentUser;
+            var config = new FlexibleSyncConfiguration(user);
+            taskRealm = Realm.GetInstance(config);
+            AddSubscriptionsToRealm();
+            Session.Error += SessionErrorHandler();
+        }
+        private void AddSubscriptionsToRealm()
+        {
+           var subscriptions = taskRealm.Subscriptions;
+           subscriptions.Update(() =>
+           {
+               var defaultSubscription = taskRealm.All<Task>()
+                   .Where(t => t.OwnerId == user.Id);
+               subscriptions.Add(defaultSubscription);
+           });
+        }
+        protected override async void OnAppearing()
+        {
+            base.OnAppearing();
+            WaitingLayout.IsVisible = true;
+            try
+            {
+                SetUpTaskList();
+            }
+            catch (Exception ex)
+            {
+                await DisplayAlert("Error Fetching Tasks", ex.Message, "OK");
+            }
+        }
+
+        private void SetUpTaskList()
+        {
+            if (_tasks == null || _tasks.Count() == 0)
```

Comment:
I'm not sure how useful is this part. The loading does not happen when we call `taskRealm.All`.

If you want to do it you should do it when calling `WaitForSynchronizationAsync`